tests/test_constraints.py

```diff
@@ -47,3 +51,25 @@ def test_apply_constraint_to_req_with_extras():
     new_req, constraint = c.get_constrained_requirement(old_req)
     assert new_req == Requirement("foo[bar]==1.1")
     assert constraint == Requirement("foo==1.1")
+
+
+def test_load_empty_constraints_file():
+    assert constraints.load(None)._data == {}
+
+
+def test_load_non_existant_constraints_file(tmp_path: pathlib.Path):
+    non_existant_file = tmp_path / "non_existant.txt"
+    with pytest.raises(FileNotFoundError):
+        constraints.load(non_existant_file)
+
+
+@patch("fromager.requirements_file.parse_requirements_file")
+def test_load_constraints_file(
+    parse_requirements_file: typing.Callable, tmp_path: pathlib.Path
+):
+    constraint_file = tmp_path / "constraint.txt"
+    constraint_file.write_text("a\n")
+    parse_requirements_file.return_value = {str(constraint_file): ["torch==3.1.0"]}
```

Comment:
This no longer matches the API of the function.

Mocking fewer things and writing test content to real files will help expose any other cases like that.

Reply:
For unit tests, isn't better to mock `parse_requirements_file` so that if there is a test failure in `test_constraints.py` then we know it is because of the function we are testing and not cause some other function that it depends on?

Reply:
Sometimes, depending on how complex the stack is. But this is a pretty shallow stack of relatively simple functions, so my inclination would be to mock little.

If you can mock it in a way that the API change would have been caught, that would be OK.